plugins/aws/cloudtrail/cloudtrailObjectLock.js

```diff
@@ -0,0 +1,81 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Object Lock Enabled',
+    category: 'CloudTrail',
+    description: 'Ensures that AWS CloudTrail S3 buckets use Object Lock for data protection and regulatory compliance.',
+    more_info: 'CloudTrail buckets should be configured to have object lock enabled. You can use it to prevent an object from being deleted or overwritten for a fixed amount of time or indefinitely.',
+    recommended_action: 'Edit trail to use a bucket with object locking enabled.',
+    link: 'https://docs.aws.amazon.com/AmazonS3/latest/dev/object-lock-managing.html',
+    apis: ['CloudTrail:describeTrails', 'S3:getObjectLockConfiguration'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        async.each(regions.cloudtrail, function(region, rcb){
+
+            var describeTrails = helpers.addSource(cache, source,
+                ['cloudtrail', 'describeTrails', region]);
+
+            if (!describeTrails) return rcb();
+
+            if (describeTrails.err || !describeTrails.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query for CloudTrail policy: ' + helpers.addError(describeTrails), region);
+                return rcb();
+            }
+
+            if (!describeTrails.data.length) {
+                helpers.addResult(results, 0, 'No S3 buckets to check', region);
+                return rcb();
+            }
+
+            async.each(describeTrails.data, function(trail, cb){
+                if (!trail.S3BucketName) return cb();
+                // Skip CloudSploit-managed events bucket
+                if (trail.S3BucketName == helpers.CLOUDSPLOIT_EVENTS_BUCKET) return cb();
+
+                var s3Region = helpers.defaultRegion(settings);
+                var resource = 'arn:aws:s3:::' + trail.S3BucketName;
+                var getObjectLockConfiguration = helpers.addSource(cache, source,
+                    ['s3', 'getObjectLockConfiguration', s3Region, trail.S3BucketName]);
+
+                if (!getObjectLockConfiguration) {
+                    helpers.addResult(results, 3,
+                        'Error querying for object lock configuration for bucket: ' + trail.S3BucketName + ': ' + helpers.addError(getObjectLockConfiguration),
+                        region, resource);
+
+                    return cb();
+                }
+
+                if (getObjectLockConfiguration.err || !getObjectLockConfiguration.data) {
```

Comment:
`getObjectLockConfiguration.err` should be added to the error check above (line 46)

Reply:
So if the object lock is not enabled for s3 bucket, getObjectLockConfiguration API sends an error in response, not an empty data object. Which means that object lock is not enabled for the bucket and we should add the failed result.